Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino

```diff
@@ -337,34 +321,82 @@ void doPublishOdom() {
   pose.theta = normalize_angle(pose.theta + delta_th);
   
   // Publish the odometry message over ROS
-  ros_odom_msg.header.stamp = current_time;
+  ros_raw_odom_msg.header.stamp = current_time;
 
   // set the position
-  ros_odom_msg.pose = pose;
+  ros_raw_odom_msg.pose = pose;
 
   // set the velocity
-  ros_odom_msg.twist.vx = vx;
-  ros_odom_msg.twist.vy = 0.0;
-  ros_odom_msg.twist.vtheta = vth;
+  ros_raw_odom_msg.twist.vx = vx;
+  ros_raw_odom_msg.twist.vy = 0.0;
+  ros_raw_odom_msg.twist.vtheta = vth;
   
-  ros_odom_pub.publish(&ros_odom_msg);
+  ros_raw_odom_pub.publish(&ros_raw_odom_msg);
   
   // Remember current time for next publish
-  last_odom_publish_time = current_time;
+  last_raw_odom_publish_time = current_time;
 }
 
 /*
  * Publish imu data to ros.
  */
-void doPublishImu() {
-  // TODO(mwomack)
+void doPublishRawImu() {
+  ros::Time current_time = ros_nh.now();
+  imu.read();
+  mag.read();
+
+  // Publish the imu message over ROS
+  ros_raw_imu_msg.header.stamp = current_time;
+  // TODO(mwomack): set header frame to what?
+
+  ros_raw_imu_msg.accelerometer.x = convertLSM6ToMetersPerSecondSquared(imu.a.x);
+  ros_raw_imu_msg.accelerometer.y = convertLSM6ToMetersPerSecondSquared(imu.a.y);
+  ros_raw_imu_msg.accelerometer.z = convertLSM6ToMetersPerSecondSquared(imu.a.z);
+  
+  ros_raw_imu_msg.gyro.x = convertLSM6ToRadiansPerSecond(imu.g.x);
+  ros_raw_imu_msg.gyro.y = convertLSM6ToRadiansPerSecond(imu.g.y);
+  ros_raw_imu_msg.gyro.z = convertLSM6ToRadiansPerSecond(imu.g.z);
+  
+  ros_raw_imu_msg.magnetometer.x = convertLIS3MDLToTesla(mag.m.x);
+  ros_raw_imu_msg.magnetometer.y = convertLIS3MDLToTesla(mag.m.y);
+  ros_raw_imu_msg.magnetometer.z = convertLIS3MDLToTesla(mag.m.z);
+
+  ros_raw_imu_pub.publish(&ros_raw_imu_msg);
 }
 
-/*
- * Publish magnetometer data to ros.
+/**
+ * Converts accelerometer readings from LSM6 set to default settings
+ * in m/s^2.
  */
-void doPublishMagnetometer() {
-  // TODO(mwomack)
+double convertLSM6ToMetersPerSecondSquared(int16_t reading) {
```

Comment:
If you wanted, you could convert this to 
`geometry_msgs::Vector3 convertToMetersPerSecondSquared(const LSM6::vector<int16_t>& accel)`
or even
`geometry_msgs::Vector3 getAccelerations(const LSM6& imu)`
or something like that.
Just the general feeling of adding more type information to the functions to enforce what's passed into the them and reduce to total number of function calls.